runtime/parser2/lexer/state.go

```diff
@@ -290,6 +292,13 @@ func identifierState(l *lexer) stateFn {
 
 func stringState(l *lexer) stateFn {
 	l.scanString('"')
+	memoryGauge := l.memoryGauge
+	if memoryGauge != nil {
+		memoryGauge.UseMemory(common.MemoryUsage{
+			Kind:   common.MemoryKindString,
+			Amount: uint64(l.wordLength()),
+		})
+	}
```

Comment:
I was hesitant to do that here, actually even in the interpreter, because it would change the semantics, i.e. it would always allocate the memory usage value, even if metering is disabled (there is no gauge)